python/GafferOSLUI/OSLCodeUI.py

```diff
@@ -311,10 +311,85 @@ def __plugPopupMenu( menuDefinition, plugValueWidget ) :
+			( "/Math/Trigonometry/Tangent", "tan( angleInRadians )" ),
+			( "/Math/Trigonometry/Arc Sin", "asin( y )" ),
+			( "/Math/Trigonometry/Arc Cosine", "acos( x )" ),
+			( "/Math/Trigonometry/Arc Tangent", "atan( yOverX )" ),
+			( "/Math/Trigonometry/Arc Tangent 2", "atan2( y, x )" ),
+			( "/Math/Exponents/Pow", "pow( x, y )" ),
+			( "/Math/Exponents/Exp", "exp( x )" ),
+			( "/Math/Exponents/Log", "log( x )" ),
+			( "/Math/Exponents/Square Root", "sqrt( x )" ),
+			( "/Math/Utility/Abs", "abs( x )" ),
+			( "/Math/Utility/Sign", "sign( x )" ),
+			( "/Math/Utility/Floor", "floor( x )" ),
+			( "/Math/Utility/Ceil", "ceil( x )" ),
+			( "/Math/Utility/Round", "round( x )" ),
+			( "/Math/Utility/Trunc", "trunc( x )" ),
+			( "/Math/Utility/Mod", "mod( x )" ),
+			( "/Math/Utility/Min", "min( a, b )" ),
+			( "/Math/Utility/Max", "max( a, b )" ),
+			( "/Math/Utility/Clamp", "clamp( x, minValue, maxValue )" ),
+			( "/Math/Utility/Mix", "mix( a, b, alpha )" ),
+			( "/Math/Geometry/Dot", "dot( a, b )" ),
+
+			( "/Geometry/Cross", "cross( a, b )" ),
+			( "/Geometry/Length", "length( V )" ),
+			( "/Geometry/Length", "distance( p0, p1 )" ),
+			( "/Geometry/Normalize", "normalize( V )" ),
+			( "/Geometry/Face Forward", "faceforward( N, I )" ),
+			( "/Geometry/Reflect", "reflect( I, N )" ),
+			( "/Geometry/Refract", "refract( I, N, eta )" ),
+			( "/Geometry/Rotate", "rotate( p, angle, p0, p1 )" ),
+			( "/Geometry/Transform", "transform( toSpace, p )" ),
+			( "/Geometry/Transform", "transform( fromSpace, toSpace, p )" ),
+
+			( "/Color/Luminance", "luminance( c )" ),
+			( "/Color/BlackBody", "blackbody( degreesKelvin )" ),
+			( "/Color/Wavelength Color", "wavelength_color( wavelengthNm )" ),
+			( "/Color/Transform", "transformc( fromSpace, toSpace, c )" ),
+
+			( "/Pattern/Step", "step( edge, x )" ),
+			( "/Pattern/Linear Step", "linearstep( edge0, edge1, x )" ),
+			( "/Pattern/Smooth Step", "smoothstep( edge0, edge1, x )" ),
+			( "/Pattern/Noise", "noise( \"perlin\", p )" ),
+			( "/Pattern/Periodic Noise", "noise( \"perlin\", p, period )" ),
+			( "/Pattern/Cell Noise", "cellnoise( p )" ),
+
+			( "/String/Length", "length( str )" ),
+			( "/String/Format", "format( \"\", ... )" ),
+			( "/String/Join", "concat( str0, str1 )" ),
+			( "/String/Split", "split( str, results )" ),
+			( "/String/Starts With", "startswith( str, prefix )" ),
+			( "/String/Ends With", "endswith( str, suffix )" ),
+			( "/String/Substring", "substr( str, start, length )" ),
+			( "/String/Get Char", "getchar( str, n )" ),
+			( "/String/Hash", "hash( str )" ),
+
+			( "/Texture/Texture", "texture( filename, s, t )" ),
+			( "/Texture/Environment", "environment( filename, R )" ),
+
+		] ) :
+
+			menuDefinition.prepend( "/InsertDivider", { "divider" : True } )
+
+			menuDefinition.prepend(
+				"/Insert" + label,
+				{ "command" : functools.partial( plugValueWidget.textWidget().insertText, text ) },
```

Comment:
Yeah, hardcoding it kindof sucked, but I didn't know any other way - I don't think it can be introspected. Another thing I wanted was to have a handy reference section in the manual with all the functions and their descriptions from the osl language spec. We do link to the actual spec itself, but there's so much verbiage to get past before getting to the function reference that I fear it'll put people off.

One approach would be to crib everything out of the language spec, convert it to markdown and store it somewhere (maybe even as metadata in gaffer?). Then we could generate the reference manual and the menus from it. That seemed a bit of an effort though, so I left it as future work.

I've added `isconnected()` and `isconstant()` for good measure.

I haven't added `inFloat()` etc for a couple of reasons. Firstly, I was hoping to find the time at some point  to explore the possibility of having the inputs/outputs map directly to primvars/channels without needing the functions. Secondly, folks can always use the InFloat shaders etc connected to the inputs/outputs, which kindof encourages making a more modular shader in the middle anyway. Not a big deal to add them if you think it's definitely better that may, but I kindof wanted to see how things panned out first.